.github/workflows/tests.yml

```diff
@@ -45,7 +45,24 @@ jobs:
       # https://github.com/zf1s/zf1/pull/6#issuecomment-495397170
       - name: Validate composer.json for all packages
         run: |
-          for json in packages/*/composer.json; do COMPOSER=$json composer validate || touch failed; done
+          for json in composer.json packages/*/composer.json; do COMPOSER=$json composer validate || touch failed; done
           test ! -f failed
 
+      - name: Cache Composer packages
+        id: composer-cache
+        uses: actions/cache@v2
+        with:
+          path: vendor
+          # hashFiles is path relative to GITHUB_WORKSPACE
+          # we hash only root composer.json
+          # https://docs.github.com/en/free-pro-team@latest/actions/reference/context-and-expression-syntax-for-github-actions#hashfiles
+          # https://docs.github.com/en/free-pro-team@latest/actions/reference/environment-variables#default-environment-variables
+          key: ${{ runner.os }}-php-${{ hashFiles('/composer.json') }}
```

Comment:
now I'm thinking perhaps the different php versions should not share composer cache:
- https://github.com/zf1s/zf1/actions/runs/461660032

otherwise, they end up not being ran at all, as cache created from other run, which is possibly different PHP version... which could end up shadowing some bugs.

but then again vendor should be identical in regardless what was the PHP version when composer install was invoked. there's one gotcha: composer v2 generates different autoload if [platform-check](https://getcomposer.org/doc/06-config.md#platform-check) is enabled (and it is by default).


@falkenhawk wdyt?

Reply:
made the change to be able to proceed with merging. can easily remove the commit if it's unwanted.